system/lib/wasmfs/syscalls.cpp

```diff
@@ -20,11 +20,50 @@
 #include <utility>
 #include <vector>
 #include <wasi/api.h>
+#include <unistd.h>
 
 extern "C" {
 
 using namespace wasmfs;
 
+// Copy the file specified by the pathname into JS.
+// Return a pointer to the JS buffer in HEAPU8.
```

Comment:
It would be good to document that the buffer should be `free`d by the caller as well.